src/module_2/groceries_analysis.md

```diff
@@ -0,0 +1,1811 @@
+
+
+```python
+# Calculate order size as the length of the ordered_items list for each order
+orders_df['order_size'] = orders_df['ordered_items'].apply(len)
+# Analyze the distribution of order sizes
+order_size_distribution = orders_df['order_size'].describe()
+display(order_size_distribution)
+printmd(f"Most common order size: {orders_df['order_size'].mode()[0]}")
+plt.hist(orders_df['order_size'], bins=50, edgecolor='k')
+plt.title("Distribution of Orders Sizes")
+plt.xlabel("Order Size (Number of Items)")
+plt.ylabel("Frequency")
+plt.show()
+```
+
+
+    count    8773.000000
+    mean       12.305711
+    std         6.839507
+    min         1.000000
+    25%         8.000000
+    50%        11.000000
+    75%        15.000000
+    max       114.000000
+    Name: order_size, dtype: float64
+
+
+
+Most common order size: 10
+
+
+
+    
+![png](groceries_analysis_files/groceries_analysis_11_2.png)
+    
+
+
+As we can see, most orders have 10 items, but there are users who make orders with lots of items creating a long right tail. 
+Let's now check the most ordered items. 
+As each item has a unique id, we can't know what each item is. However, we can see that all items fall into a specific category, thus ,we can show the most ordered items by mapping its id to its category.
+
+
+```python
+item_frequencies = orders_df.explode('ordered_items')['ordered_items'].value_counts().reset_index().rename(columns={'ordered_items':'variant_id','count':'item_count'})
+
+# merge with inventory to get category information
+common_items = pd.merge(item_frequencies, inventory_df, on='variant_id', how='inner')
+
+# plot the top 10 most ordered items
+plt.bar(common_items['variant_id'][:10].astype(str),common_items['item_count'][:10],edgecolor='k')
+plt.title('Top 10 Most Ordered Items')
+plt.xlabel('Item')
+plt.ylabel('Frequency')
+plt.xticks(ticks=range(0,10),labels=common_items['product_type'][:10],rotation=45, ha='right')
+plt.show()
+
+```
+
+
+    
+![png](groceries_analysis_files/groceries_analysis_13_0.png)
+    
+
+
+There is a catch here, as we don't have the category of each item, when we merged the datasets, item_frequencies(n=2117) and inventory(n=1733), we lost some items. This is because the inventory dataset has less items than the orders dataset.
+
+
+```python
+# Determine the most commonly ordered categories
+common_categories = common_items.groupby('product_type')['item_count'].sum().reset_index().sort_values('item_count', ascending=False)
+
+plt.bar(x=common_categories['product_type'][:10], height=common_categories['item_count'][:10], edgecolor='k')
+plt.title('Top 10 Most Ordered Categories')
+plt.xlabel('Product Category')
+plt.ylabel('Frecuency')
+plt.xticks(rotation=45, ha = 'right')  
+plt.show()
+```
+
+
+    
+![png](groceries_analysis_files/groceries_analysis_15_0.png)
+    
+
+
+
+```python
+# Top 10 most ordered brands
```

Comment:
It seems you are repeating always the same piece of code. You could consider make a function and call it with different arguments